fiftyone/utils/hf_hub.py

```diff
@@ -0,0 +1,723 @@
+    label_fields = config.label_fields
+    if label_fields is None:
+        return label_field_names, label_types
+
+    for label_type, fields in label_fields.items():
+        if isinstance(fields, str):
+            label_field_names.append(fields)
+            label_types.append(label_type)
+        elif isinstance(fields, list):
+            label_field_names.extend(fields)
+            label_types.extend([label_type] * len(fields))
+
+    return label_field_names, label_types
+
+
+def _get_parquet_dataset_features(repo_id, split, subset, revision=None):
+    api_url = f"{DATASETS_SERVER_URL}/info?dataset={repo_id.replace('/', '%2F')}&config={subset}&split={split}]"
+    if revision is not None:
+        api_url += f"&revision={revision}"
+
+    response = requests.get(api_url)
+    features = response.json()["dataset_info"]["features"]
+    return features
+
+
+def _get_num_rows(repo_id, split, subset, revision=None):
+    api_url = f"{DATASETS_SERVER_URL}/info?dataset={repo_id.replace('/', '%2F')}&config={subset}&split={split}]"
+    if revision is not None:
+        api_url += f"&revision={revision}"
+
+    response = requests.get(api_url)
+    splits = response.json()["dataset_info"]["splits"]
+    return splits[split]["num_examples"]
+
+
+def _build_rows_request_url(
+    repo_id, split=None, subset="default", revision=None, offset=0, length=100
+):
+    url = f"{DATASETS_SERVER_URL}/rows?dataset={repo_id.replace('/', '%2F')}"
+    if split is not None:
+        url += f"&split={split}"
+    if subset is not None:
+        url += f"&config={subset}"
+    if revision is not None:
+        url += f"&revision={revision}"
+    url += f"&offset={offset}&length={length}"
+    return url
+
+
+def _get_rows(
+    repo_id, split, subset, start_index=0, end_index=100, revision=None
+):
+    length = end_index - start_index
+    url = _build_rows_request_url(
+        repo_id, split, subset, revision, offset=start_index, length=length
+    )
+    response = requests.get(url)
+    return response.json()["rows"]
+
+
+def _build_media_field_converter(
+    media_field_key, media_field_name, feature, download_dir
+):
+    def convert_media_field(sample_dict, row):
+        row_content = row["row"]
+        row_index = row["row_idx"]
+
+        filename = f"{media_field_name}_{row_index}.png"
+        filepath = os.path.join(download_dir, filename)
+
+        if feature["_type"] == "Image":
+            url = row_content[media_field_name]["src"]
+        else:
+            url = row_content[media_field_name]
+
+        sample_dict[media_field_key] = filepath
+        try:
+            ## create the file if it doesn't exist
+            if not os.path.exists(filepath):
+                Image.open(requests.get(url, stream=True).raw).save(filepath)
+        except:
+            logger.warning(f"Failed to download image from {url}")
+
+    return convert_media_field
+
+
+def _get_image_shape(image_path):
+    with Image.open(image_path) as img:
```

Comment:
please use this instead to avoid reimplementation (also it handles some tricky cases like EXIF tags):

```py
import fiftyone.core.metadata as fom

def _get_image_shape(image_path):
    metadata = fom.ImageMetadata.build_for(image_path)
    return (metadata.width, metadata.height)
```

Reply:
Can do!